packages/dht/test/unit/StoreRpcLocal.test.ts

```diff
@@ -0,0 +1,167 @@
+import { range } from 'lodash'
+import { StoreRpcLocal } from '../../src/dht/store/StoreRpcLocal'
+import { areEqualPeerDescriptors, createRandomDhtAddress, DhtAddress, getDhtAddressFromRaw, getRawFromDhtAddress } from '../../src/identifiers'
+import { DataEntry, PeerDescriptor, StoreDataRequest } from '../../src/proto/packages/dht/protos/DhtRpc'
+import { createMockPeerDescriptor } from '../utils/utils'
+import { getClosestNodes } from '../../src/dht/contact/getClosestNodes'
+import { wait } from '@streamr/utils'
+
+describe('StoreRpcLocal', () => {
+
+    const NODE_COUNT = 5
+    const DATA_ENTRY = {
+        key: getRawFromDhtAddress(createRandomDhtAddress()),
+        creator: getRawFromDhtAddress(createRandomDhtAddress())
+    }
+
+    const ALL_NODES = range(NODE_COUNT).map(() => createMockPeerDescriptor())
+
+    const getNodeCloseToData = (distanceIndex: number) => {
+        const dataKey = getDhtAddressFromRaw(DATA_ENTRY.key)
+        return getClosestNodes(dataKey, ALL_NODES)[distanceIndex]
+    }
+
+    let storeRpcLocal: StoreRpcLocal
+    let setAllEntriesAsStale: jest.Mock<undefined, [DhtAddress]>
+    let replicateDataToContact: jest.Mock<Promise<void>, [DataEntry, PeerDescriptor]>
+
+    beforeEach(() => {
+        setAllEntriesAsStale = jest.fn<undefined, [DhtAddress]>()
+        replicateDataToContact = jest.fn<Promise<void>, [DataEntry, PeerDescriptor]>()
+    })
+
+    describe('local node is primary storer', () => {
+
+        beforeEach(() => {
+            const localPeerDescriptor = getNodeCloseToData(0)
+            storeRpcLocal = new StoreRpcLocal({
+                localDataStore: { 
+                    storeEntry: () => true,
+                    setAllEntriesAsStale
+                } as any,
+                localPeerDescriptor,
+                replicateDataToContact,
+                getStorers: () => getClosestNodes(getDhtAddressFromRaw(DATA_ENTRY.key), ALL_NODES)
+            })
+        })
+
+        it('storeData', async () => {
+            const request = StoreDataRequest.create({
+                key: DATA_ENTRY.key,
+            })
+            await storeRpcLocal.storeData(request)
+            expect(setAllEntriesAsStale).not.toHaveBeenCalled()
+        })
+
+        it('replicateData', async () => {
+            const request = {
+                entry: DATA_ENTRY
+            }
+            await storeRpcLocal.replicateData(request as any, { incomingSourceDescriptor: createMockPeerDescriptor() } as any)
+            expect(setAllEntriesAsStale).not.toHaveBeenCalled()
+            // Wait for setImmediate
+            await wait(50)
+            expect(replicateDataToContact).toHaveBeenCalledTimes(4)
```

Comment:
Could calculate 4 from `NODE_COUNT` (if we don't want explicitly assert the payload of the `replicateDataToContact` calls)

```suggestion
            // All other nodes except local node
            expect(replicateDataToContact).toHaveBeenCalledTimes(NODE_COUNT - 1)
```